datalad/distribution/dataset.py

```diff
@@ -141,6 +144,41 @@ def id(self):
 
         return self._id
 
+    @property
+    def config(self):
+        """Get an instance of the parser for the persistent dataset configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfg is None:
+            cfgpath = opj(self.path, '.datalad', 'config')
+            if not exists(cfgpath):
+                return None
+            self._cfg = git.GitConfigParser(cfgpath, read_only=True)
+        return self._cfg
+
+    @property
+    def config_writer(self):
+        """Get an instance of the parser for modifying the persistent dataset
+        configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfgwriter is None:
+            cfgpath = opj(self.path, '.datalad')
+            if not exists(cfgpath):
+                os.makedirs(cfgpath)
+            self._cfgwriter = git.GitConfigParser(
+                opj(cfgpath, 'config'),
+                read_only=False,
+                merge_includes=False)
+        return self._cfgwriter
```

Comment:
I think it is somewhat "inspired" by GitPython.  Why not to have `.config` and just do saving right upon `set` and may be just allow for `with ds.config:` batch mode where changes get flushed after leaving the context block?